c_cxx/squeezenet/main.cpp

```diff
@@ -5,6 +5,8 @@
 #include <assert.h>
 #include <vector>
 #include <onnxruntime_cxx_api.h>
+
+#ifdef HAVE_TENSORRT_PROVIDER_FACTORY_H
 #include <tensorrt_provider_factory.h>
 #include <tensorrt_provider_options.h>
```

Comment:
Quick question here.
In line 41, the OrtTensorRTProviderOptionsV2 is defined in tensorrt_provider_options.h
It seems line 41 also should be in #ifdef HAVE_TENSORRT_PROVIDER_FACTORY_H ?

Reply:
Interesting. Why it could work now?

Reply:
SessionOptionsAppendExecutionProvider_TensorRT_V2 is declared in onnxruntime_c_api.h, it works similar. onnxruntime_c_api.h doesn't include tensorrt_provider_options.h.  So I think it is ok.

Reply:
Oh, in onnxruntime_c_api.h we have "ORT_RUNTIME_CLASS(TensorRTProviderOptionsV2);" . So it is enough.